include/nuttx/pcnt/pcntchar.h

```diff
@@ -20,80 +20,69 @@
 
-#include <errno.h>
-#include <debug.h>
-#include <stdio.h>
+#include <sys/types.h>
+#include <stdint.h>
+#include <stdbool.h>
 
-#include <nuttx/sensors/qencoder.h>
-#include <arch/board/board.h>
+#include <nuttx/pcnt/pcnt.h>
 
-#include "espressif/esp_qencoder.h"
+#ifdef CONFIG_PCNTCHAR
 
 /****************************************************************************
- * Public Functions
+ * Pre-processor Definitions
  ****************************************************************************/
 
 /****************************************************************************
- * Name: board_qencoder_initialize
- *
- * Description:
- *   Initialize the quadrature encoder driver
- *
+ * Public Types
  ****************************************************************************/
 
-int board_qencoder_initialize(void)
-{
-  int ret;
-  char devpath[12];
-  int devno = 0;
+/****************************************************************************
+ * Public Data
+ ****************************************************************************/
 
-  /* Initialize a quadrature encoder interface. */
-#ifdef CONFIG_ESP_PCNT_U0_QE
-  snprintf(devpath, sizeof(devpath), "/dev/qe%d", devno++);
-  ret = esp_qeinitialize(devpath, 0);
-  if (ret < 0)
-    {
-      syslog(LOG_ERR, "ERROR: esp_qeinitialize failed: %d\n", ret);
-      return ret;
-    }
+#undef EXTERN
+#if defined(__cplusplus)
+#define EXTERN extern "C"
+extern "C"
+{
+#else
+#define EXTERN extern
 #endif
 
-#ifdef CONFIG_ESP_PCNT_U1_QE
-  snprintf(devpath, sizeof(devpath), "/dev/qe%d", devno++);
-  ret = esp_qeinitialize(devpath, 1);
-  if (ret < 0)
-    {
-      syslog(LOG_ERR, "ERROR: esp_qeinitialize failed: %d\n", ret);
-      return ret;
-    }
-#endif
+/****************************************************************************
+ * Public Functions Prototypes
+ ****************************************************************************/
 
-#ifdef CONFIG_ESP_PCNT_U2_QE
-  snprintf(devpath, sizeof(devpath), "/dev/qe%d", devno++);
-  ret = esp_qeinitialize(devpath, 2);
-  if (ret < 0)
-    {
-      syslog(LOG_ERR, "ERROR: esp_qeinitialize failed: %d\n", ret);
-      return ret;
-    }
-#endif
+/****************************************************************************
+ * Name: pcntchar_register
+ *
+ * Description:
+ *   Create and register the pulse counter (pcnt) character driver.
+ *
+ * Input Parameters:
+ *   pcnt - An instance of the lower half pcnt driver
+ *
+ * Returned Value:
+ *   OK if the driver was successfully registered; A negated errno value is
+ *   returned on any failure.
+ *
+ ****************************************************************************/
 
-#ifdef CONFIG_ESP_PCNT_U3_QE
-  snprintf(devpath, sizeof(devpath), "/dev/qe%d", devno++);
-  ret = esp_qeinitialize(devpath, 3);
-  if (ret < 0)
-    {
-      syslog(LOG_ERR, "ERROR: esp_qeinitialize failed: %d\n", ret);
-      return ret;
-    }
-#endif
+int pcntchar_register(FAR struct pcnt_dev_s *pcnt);
```

Comment:
`FAR const char *path` should be in register arguments, like its done for other drivers like pwm, adc, dac, timer.

Reply:
It was implemented in this way on [i2c driver](https://github.com/apache/nuttx/blob/master/drivers/i2c/i2c_driver.c#L394). Not sure but seems this might also valid.

Reply:
I2C is a bus driver and for some reason BUS drivers have a different convention. Probably because the system tools from apps only work with a given devpath (i2ctools and spitools). Most device drivers use `path` in arguments, I didn't check if all of them. Details about bus drivers vs device drivers are here https://nuttx.apache.org/docs/latest/implementation/device_drivers.html#bus-drivers

Reply:
Up to you but we can keep it for now because this driver will be there for temporary.